### PR TITLE
Implement API AppendAllBytes

### LINKED ISSUE #84532: [API Proposal]: File.AppendAllBytes
### Background and motivation

The File class has all methods for bytes which also exist for text except of appending data. Not only that it fits into the api surface by filling this gap is also saves us writing more than one line of code.

### API Proposal

```diff
namespace System.IO;

public class File
{
+    public static void AppendAllBytes(string path, ReadOnlySpan<byte> data);
+    public static Task AppendAllBytesAsync(string path, ReadOnlyMemory<byte> data, CancellationToken cancellationToken = default);
     public static void AppendAllText(string path, string? contents);
     public static Task AppendAllTextAsync(string path, string? contents, CancellationToken cancellationToken = default);
}
```

### API Usage

```csharp

byte[] data = File.ReadAllBytes("C:\\1.bin");

File.AppendAllBytes("C:\\2.bin", data);


```

### Alternative Designs

_No response_

### Risks

_No response_

### PR DESCRIPTION
**Pull request by MojtabaTajik**
Added AppendAllBytes API and the async version with the corresponding tests.

Fix #84532

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 1]
```diff
@@ -689,6 +689,63 @@ public static void WriteAllBytes(string path, byte[] bytes)
             RandomAccess.WriteAtOffset(sfh, bytes, 0);
         }
 
+        /// <summary>
+        /// Appends the specified byte array to the end of the file at the given path.
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:693]:
```suggestion
        /// Appends the specified byte array to the end of the file at the given path.
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 2]
```diff
@@ -689,6 +689,63 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
         }
 
+        /// <summary>
+        /// Appends the specified byte array to the end of the file at the given path. 
+        /// If the file doesn't exist, this method creates a new file.
+        /// </summary>
+        /// <param name="path">The path of the file to which the bytes should be appended.</param>
+        /// <param name="bytes">The byte array to append to the file.</param>
+        /// <exception cref="T:System.ArgumentException">
+        /// Thrown when <paramref name="path" /> is a zero-length string, contains invalid characters
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:699]:
```suggestion
        /// Thrown when <paramref name="path" /> is a zero-length string, contains invalid characters
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 3]
```diff
@@ -689,6 +689,63 @@ public static void WriteAllBytes(string path, byte[] bytes)
             RandomAccess.WriteAtOffset(sfh, bytes, 0);
         }
 
+        /// <summary>
+        /// Appends the specified byte array to the end of the file at the given path. 
+        /// If the file doesn't exist, this method creates a new file.
+        /// </summary>
+        /// <param name="path">The path of the file to which the bytes should be appended.</param>
+        /// <param name="bytes">The byte array to append to the file.</param>
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:697]:
```suggestion
        /// <param name="bytes">The bytes to append to the file.</param>
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 4]
```diff
@@ -689,6 +689,63 @@ public static void WriteAllBytes(string path, byte[] bytes)
             RandomAccess.WriteAtOffset(sfh, bytes, 0);
         }
 
+        /// <summary>
+        /// Appends the specified byte array to the end of the file at the given path. 
+        /// If the file doesn't exist, this method creates a new file.
+        /// </summary>
+        /// <param name="path">The path of the file to which the bytes should be appended.</param>
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:696]: Follow existing conventions.
```suggestion
        /// <param name="path">The file to append to.</param>
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 5]
```diff
@@ -689,6 +689,63 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// <param name="bytes">The byte array to append to the file.</param>
+        /// <exception cref="T:System.ArgumentException">
+        /// Thrown when <paramref name="path" /> is a zero-length string, contains invalid characters 
+        /// (which can be queried using the <see cref="M:System.IO.Path.GetInvalidPathChars" /> method), 
+        /// or when <paramref name="bytes" /> is null or empty.
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:701]:
```suggestion
        /// <paramref name="path" /> is a zero-length string, contains invalid characters
        /// (which can be queried using the <see cref="M:System.IO.Path.GetInvalidPathChars" /> method),
        /// or when <paramref name="bytes" /> is null or empty.
```

Comment (Jozkee) [reply src/libraries/System.Private.CoreLib/src/System/IO/File.cs:701]: > when <paramref name="bytes" /> is null or empty

We should not throw when `bytes` is empty. I think this is just a documentation error, no?

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 6]
```diff
@@ -689,6 +689,63 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// <summary>
+        /// Asynchronously appends the specified byte array to the end of the file at the given path. 
+        /// If the file doesn't exist, this method creates a new file. If the operation is canceled, the task will be returned in a canceled state.
+        /// </summary>
+        /// <param name="path">The path of the file to which the bytes should be appended.</param>
+        /// <param name="bytes">The byte array to append to the file.</param>
+        /// <param name="cancellationToken">A token to monitor for cancellation requests. The default value is <see cref="T:System.Threading.CancellationToken.None" />.</param>
+        /// <returns>A task that represents the asynchronous append operation.</returns>
+        /// <exception cref="T:System.ArgumentException">
+        /// Thrown when <paramref name="path" /> is a zero-length string, contains invalid characters
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:725]:
```suggestion
        /// Thrown when <paramref name="path" /> is a zero-length string, contains invalid characters
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 7]
```diff
@@ -689,6 +689,63 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// Asynchronously appends the specified byte array to the end of the file at the given path. 
+        /// If the file doesn't exist, this method creates a new file. If the operation is canceled, the task will be returned in a canceled state.
+        /// </summary>
+        /// <param name="path">The path of the file to which the bytes should be appended.</param>
+        /// <param name="bytes">The byte array to append to the file.</param>
+        /// <param name="cancellationToken">A token to monitor for cancellation requests. The default value is <see cref="T:System.Threading.CancellationToken.None" />.</param>
+        /// <returns>A task that represents the asynchronous append operation.</returns>
+        /// <exception cref="T:System.ArgumentException">
+        /// Thrown when <paramref name="path" /> is a zero-length string, contains invalid characters 
+        /// (which can be queried using the <see cref="M:System.IO.Path.GetInvalidPathChars" /> method),
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:726]:
```suggestion
        /// (which can be queried using the <see cref="M:System.IO.Path.GetInvalidPathChars" /> method),
```

File: src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+            string path = GetTestFilePath();
+
+            Assert.Throws<ArgumentNullException>(() => File.AppendAllBytes(null, new byte[0]));
+            Assert.Throws<ArgumentNullException>(() => File.AppendAllBytes(path, null));
+        }
+
+        [Fact]
+        public void InvalidParameters()
+        {
+            Assert.Throws<ArgumentException>(() => File.AppendAllBytes(string.Empty, new byte[0]));
```
Comment (Jozkee) [inline src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs:26]: Is `new byte[0]` an invalid parameter?

File: src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsFileLockingEnabled))]
+        public void OpenFile_ThrowsIOException()
+        {
+            string path = GetTestFilePath();
+            byte[] bytes = Encoding.UTF8.GetBytes("bytes");
+
+            using (File.Create(path))
+            {
+                Assert.Throws<IOException>(() => File.AppendAllBytes(path, bytes));
+                Assert.Throws<IOException>(() => File.ReadAllBytes(path));
```
Comment (Jozkee) [inline src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs:67]: This is already tested in https://github.com/dotnet/runtime/blob/00f1e0c94a3853f65d153179c1981d5b3f01c5e6/src/libraries/System.IO.FileSystem/tests/File/ReadWriteAllBytes.cs#L79
```suggestion
```

File: src/libraries/System.IO.FileSystem/tests/File/AppendAllBytesAsync.cs [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsFileLockingEnabled))]
+        public async Task OpenFile_ThrowsIOExceptionAsync()
+        {
+            string path = GetTestFilePath();
+            byte[] bytes = Encoding.UTF8.GetBytes("bytes");
+
+            using (File.Create(path))
+            {
+                await Assert.ThrowsAsync<IOException>(async () => await File.AppendAllBytesAsync(path, bytes));
+                await Assert.ThrowsAsync<IOException>(async () => await File.ReadAllBytesAsync(path));
```
Comment (Jozkee) [inline src/libraries/System.IO.FileSystem/tests/File/AppendAllBytesAsync.cs:67]:
```suggestion
```

File: src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs (hunk 8 above)
Comment (MojtabaTajik) [reply src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs:26]: Nope, the `string.Empty` for the path parameter. We have same tests for other APIs, for example [File_ReadWriteAllBytes](https://github.com/dotnet/runtime/blob/10ce9fb494c9f2a16a757e1c8a1f902c28be4381/src/libraries/System.IO.FileSystem/tests/File/ReadWriteAllBytes.cs#L25).

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs (hunk 5 above)
Comment (MojtabaTajik) [reply src/libraries/System.Private.CoreLib/src/System/IO/File.cs]: Indeed, I need to correct the documentation. The behaviour is to throw an exception for a null param, not an empty byte array.

File: src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+                {
+                    Assert.Throws<UnauthorizedAccessException>(() => File.AppendAllBytes(path, dataToAppend));
+                }
+            }
+            finally
+            {
+                File.SetAttributes(path, FileAttributes.Normal);
+            }
+        }
+
```
Comment (Jozkee) [inline src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs:100]: Whitespace
```suggestion
```

File: src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+            Assert.True(result.SequenceEqual(expectedBytes));
+        }
+
+
+        [Fact]
+        public void EmptyContentCreatesFile()
+        {
+            string path = GetTestFilePath();
+            File.AppendAllBytes(path, new byte[0]);
+            Assert.True(File.Exists(path));
```
Comment (Jozkee) [inline src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs:54]:
```suggestion
            Assert.False(File.Exists(path));
            File.AppendAllBytes(path, new byte[0]);
            Assert.True(File.Exists(path));
```

File: src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+
+            Assert.Throws<ArgumentNullException>(() => File.AppendAllBytes(null, new byte[0]));
+            Assert.Throws<ArgumentNullException>(() => File.AppendAllBytes(path, null));
+        }
+
+        [Fact]
+        public void InvalidParameters()
+        {
+            Assert.Throws<ArgumentException>(() => File.AppendAllBytes(string.Empty, new byte[0]));
+        }
```
Comment (Jozkee) [inline src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs:27]: Can you please add a test that throws if the directory does not exist (for async too) e.g:
https://github.com/dotnet/runtime/blob/10ce9fb494c9f2a16a757e1c8a1f902c28be4381/src/libraries/System.IO.FileSystem/tests/File/ReadWriteAllText.cs#L42-L46

File: src/libraries/System.IO.FileSystem/tests/File/AppendAllBytesAsync.cs [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+                }
+            }
+            finally
+            {
+                File.SetAttributes(path, FileAttributes.Normal);
+            }
+        }
+
+    }
+
```
Comment (Jozkee) [inline src/libraries/System.IO.FileSystem/tests/File/AppendAllBytesAsync.cs:102]:
```suggestion
    }
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 15]
```diff
@@ -689,6 +689,62 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// <summary>
+        /// Appends the specified byte array to the end of the file at the given path.
+        /// If the file doesn't exist, this method creates a new file.
+        /// </summary>
+        /// <param name="path">The file to append to.</param>
+        /// <param name="bytes">The bytes to append to the file.</param>
+        /// <exception cref="System.ArgumentException">
+        /// Thrown when <paramref name="path"/> is a zero-length string, contains only white space,
+        /// or contains invalid characters as defined by <see cref="System.IO.Path.GetInvalidPathChars"/>.
+        /// </exception>
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:701]: Don't use "Throw when".
```suggestion
        /// <exception cref="System.ArgumentException">
        /// <paramref name="path"/> is a zero-length string, contains only white space, or contains one more invalid characters defined by the <see cref="System.IO.Path.GetInvalidPathChars"/> method.
        /// </exception>
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 16]
```diff
@@ -689,6 +689,62 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// <exception cref="System.ArgumentNullException">
+        /// Thrown when <paramref name="path"/> is null, or when <paramref name="bytes"/> is null.
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:703]:
```suggestion
        /// Either <paramref name="path"/> or <paramref name="bytes"/> is null.
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 17]
```diff
@@ -689,6 +689,62 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+            using SafeFileHandle fileHandle = OpenHandle(path, FileMode.Append, FileAccess.Write, FileShare.Read);
+            long fileOffset = RandomAccess.GetLength(fileHandle);
+            RandomAccess.WriteAtOffset(fileHandle, bytes, fileOffset);
+        }
+
+        /// <summary>
+        /// Asynchronously appends the specified byte array to the end of the file at the given path.
+        /// If the file doesn't exist, this method creates a new file. If the operation is canceled, the task will be returned in a canceled state.
+        /// </summary>
+        /// <param name="path">The path of the file to which the bytes should be appended.</param>
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:719]:
```suggestion
        /// <param name="path">The file to append to.</param>
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 18]
```diff
@@ -689,6 +689,62 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// <param name="bytes">The byte array to append to the file.</param>
+        /// <param name="cancellationToken">A token to monitor for cancellation requests. The default value is <see cref="T:System.Threading.CancellationToken.None" />.</param>
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:721]:
```suggestion
        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="T:System.Threading.CancellationToken.None" />.</param>
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 19]
```diff
@@ -689,6 +689,62 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// <returns>A task that represents the asynchronous append operation.</returns>
+        /// <exception cref="T:System.ArgumentException">
+        /// Thrown when <paramref name="path" /> is a zero-length string, contains invalid characters
+        /// (which can be queried using the <see cref="M:System.IO.Path.GetInvalidPathChars" /> method),
+        /// or when <paramref name="bytes" /> is null or empty.
+        /// </exception>
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:727]:
```suggestion
        /// <exception cref="T:System.ArgumentException">
        /// <paramref name="path"/> is a zero-length string, contains only white space, or contains one more invalid characters defined by the <see cref="System.IO.Path.GetInvalidPathChars"/> method.
        /// </exception>
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 20]
```diff
@@ -689,6 +689,62 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// <exception cref="T:System.ArgumentNullException">
+        /// Thrown when <paramref name="path" /> is <see langword="null" />.
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:729]:
```suggestion
        /// Either <paramref name="path"/> or <paramref name="bytes"/> is null.
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 21]
```diff
@@ -689,6 +689,62 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// </exception>
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:730]:
```suggestion
        /// </exception>
        /// <exception cref="T:System.OperationCanceledException">
        /// The cancellation token was canceled. This exception is stored into the returned task.
        /// </exception>
```

**Review by jozkee — COMMENTED** [review]
Almost there.

When you commit changes, can you please push when you have all changes ready or use the "Add suggestion to batch" in GH; each push may trigger a CI run which is more expensive than having one after all feedback is addressed.

File: src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs (hunk 13 above)
Comment (MojtabaTajik) [reply src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs:27]: Sure 👌

Comment (MojtabaTajik) [reply src/libraries/System.IO.FileSystem/tests/File/AppendAllBytes.cs:33]: Done.

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 22]
```diff
@@ -689,6 +689,60 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// Asynchronously appends the specified byte array to the end of the file at the given path.
+        /// If the file does not exist, a new file will be created. If the operation is canceled, the task will return in a canceled state.
+        /// </summary>
+        /// <param name="path">The file to append to.</param>
+        /// <param name="bytes">The bytes to append to the file.</param>
+        /// <param name="cancellationToken">A token to monitor for cancellation requests. The default value is <see cref="System.Threading.CancellationToken.None"/>.</param>
+        /// <returns>A task that represents the asynchronous append operation.</returns>
+        /// <exception cref="System.ArgumentException">
+        /// Thrown when <paramref name="path"/> is a zero-length string, contains only white space,
+        /// or contains invalid characters as defined by <see cref="System.IO.Path.GetInvalidPathChars"/>.
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:724]:
```suggestion
        /// <paramref name="path"/> is a zero-length string, contains only white space, or contains one more invalid characters defined by the <see cref="System.IO.Path.GetInvalidPathChars"/> method.
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 23]
```diff
@@ -689,6 +689,60 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// </exception>
+        /// <exception cref="System.ArgumentNullException">
+        /// Thrown when <paramref name="path"/> is null, or when <paramref name="bytes"/> is null.
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:727]:
```suggestion
        /// Either <paramref name="path"/> or <paramref name="bytes"/> is null.
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 24]
```diff
@@ -689,6 +689,60 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+            ArgumentNullException.ThrowIfNull(bytes);
+
+            using SafeFileHandle fileHandle = OpenHandle(path, FileMode.Append, FileAccess.Write, FileShare.Read);
+            long fileOffset = RandomAccess.GetLength(fileHandle);
+            RandomAccess.WriteAtOffset(fileHandle, bytes, fileOffset);
+        }
+
+        /// <summary>
+        /// Asynchronously appends the specified byte array to the end of the file at the given path.
+        /// If the file does not exist, a new file will be created. If the operation is canceled, the task will return in a canceled state.
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:716]:
```suggestion
        /// If the file doesn't exist, this method creates a new file. If the operation is canceled, the task will return in a canceled state.
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 25]
```diff
@@ -689,6 +689,60 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// </summary>
+        /// <param name="path">The file to append to.</param>
+        /// <param name="bytes">The bytes to append to the file.</param>
+        /// <param name="cancellationToken">A token to monitor for cancellation requests. The default value is <see cref="System.Threading.CancellationToken.None"/>.</param>
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:720]:
```suggestion
        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="System.Threading.CancellationToken.None"/>.</param>
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 26]
```diff
@@ -689,6 +689,60 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// <returns>A task that represents the asynchronous append operation.</returns>
+        /// <exception cref="System.ArgumentException">
+        /// Thrown when <paramref name="path"/> is a zero-length string, contains only white space,
+        /// or contains invalid characters as defined by <see cref="System.IO.Path.GetInvalidPathChars"/>.
+        /// </exception>
+        /// <exception cref="System.ArgumentNullException">
+        /// Thrown when <paramref name="path"/> is null, or when <paramref name="bytes"/> is null.
+        /// </exception>
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:728]:
```suggestion
        /// </exception>
        /// <exception cref="T:System.OperationCanceledException">
	    /// The cancellation token was canceled. This exception is stored into the returned task.
	    /// </exception>
```

**Review by jozkee — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 27]
```diff
@@ -689,6 +689,62 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="System.Threading.CancellationToken.None"/>.</param>
+        /// <returns>A task that represents the asynchronous append operation.</returns>
+        /// <exception cref="System.ArgumentException">
+        /// <paramref name="path"/> is a zero-length string, contains only white space, or contains one more invalid characters defined by the <see cref="System.IO.Path.GetInvalidPathChars"/> method.
+        /// </exception>
+        /// <exception cref="System.ArgumentNullException">
+        /// Either <paramref name="path"/> or <paramref name="bytes"/> is null.
+        /// </exception>
+        /// <exception cref="T:System.OperationCanceledException">
+	    /// The cancellation token was canceled. This exception is stored into the returned task.
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:729]:
```suggestion
        /// The cancellation token was canceled. This exception is stored into the returned task.
```

File: src/libraries/System.Private.CoreLib/src/System/IO/File.cs [hunk 28]
```diff
@@ -689,6 +689,62 @@ public static void WriteAllBytes(string path, byte[] bytes)
...
+	    /// </exception>
```
Comment (Jozkee) [inline src/libraries/System.Private.CoreLib/src/System/IO/File.cs:730]:
```suggestion
        /// </exception>
```

**Pull request merged**